client/blocks/reader-post-card/index.jsx

```diff
@@ -48,9 +51,8 @@ export default class ReaderPostCard extends React.Component {
 	};
 
 	propagateCardClick = () => {
-		// If we have an original post available (e.g. for a Discover pick), send the original post
-		// to the full post view
-		const postToOpen = this.props.originalPost ? this.props.originalPost : this.props.post;
+		// If we have an discover pick post available, send the discover pick to the full post view
+		const postToOpen = get( this.props, 'discoverPick.post' ) || this.props.post;
```

Comment:
`post.discoverPick` is initially `undefined` till we load the discover pick post or the discover site post from the stores/redux.